### PR TITLE
[WIP] src,tools: output /ayo executable, related fork impl

### PR DESCRIPTION
**Pull request by Fishrock123**
This allows you to build an `ayo` executable. Tests pass.

No windows work yet. I'm sure there are bits I forgot.

### CODE REVIEW

**Review by addaleax — APPROVED** [review]

**Review by ghost — APPROVED** [review]
did iojs have a `node` alias?

**Comment by Fishrock123** [comment]
> did iojs have a `node` alias?

Yes, I'd been thinking about that and I think the `out/Release/*` stuff is incorrect, we should use `node` there and just create a final symlink. I'll dig up iojs Makefile code for that when I have time unless someone else wants to.

**Comment by ghost** [comment]
@Fishrock123 sounds good. do you want to do the changes in this PR, or should we merge this and then create a new PR some other time?

**Comment by Fishrock123** [comment]
Please avoid merging my PRs for me in most cases - often the content is not final until I'm ready. 😅 

In this case, merging this now will only create __more__ conflicts, which is going to be a issue and there are some areas in which we should take care to make ones that wouldn't be too difficult to resolve in the future (hopefully).

**Comment by varjmes** [comment]
I put a [WIP] on this, so no-one thinks they should merge it.

**Comment by refack** [comment]
@Fishrock123 would you mind if I added changes to `vcbuild.bat`?

**Comment by Fishrock123** [comment]
@refack That'd be excellent. I'd prefer the final symlink path - I'll be changing the rest over to that tonight unless someone jumps on it before me.

**Comment by refack** [comment]
> @refack That'd be excellent. I'd prefer the final symlink path - I'll be changing the rest over to that tonight unless someone jumps on it before me.

Just so I'm clear. Build as `ayo`, then add a symlink named `node` that points to `ayo`, or the other way around?

**Comment by Fishrock123** [comment]
I have pushed up what I think will be a bit better for now - it doesn't yet produce a `node` symlink though.

**Comment by Fishrock123** [comment]
@refack have you made any progress?

I'm going to add two more commits here, particularly, adapting https://github.com/nodejs/node/commit/3f8b76ebd5599737ba99b00cfd54b5751c40b512 and https://github.com/nodejs/node/commit/72f1b348b021269f98783aee2f0e89500a20231c

**Comment by Fishrock123** [comment]
@addaleax @pup ptal

@refack maybe we can do that in another PR

File: Makefile [hunk 1]
```diff
@@ -40,9 +40,12 @@ BUILDTYPE_LOWER := $(shell echo $(BUILDTYPE) | tr '[A-Z]' '[a-z]')
 EXEEXT := $(shell $(PYTHON) -c \
 		"import sys; print('.exe' if sys.platform == 'win32' else '')")
 
-NODE_EXE = node$(EXEEXT)
+# Build to /node for easy cross-repo compat.
+BUILD_DIR_NAME = node
```
Comment (addaleax) [inline Makefile]: I have to admit I don’t quite get why this is called `DIR_NAME` and why `$(EXEEXT)` is removed? I mean, it’s still the executable name in the build dir, right…

**Comment by Fishrock123** [comment]
@addaleax good catch, should be addressed now

**Review by addaleax — APPROVED** [review]

**Comment by addaleax** [comment]
Thanks! I’d suggest merging once CI is back

(I am **so** happy we have Travis here)

**Comment by refack** [comment]
I'll follow up with a new PR for `vcbuild.bat`

**Pull request merged**